### ISSUE TITLE
panic on scroll-left attempt in EditView

### ISSUE THREAD

**Issue by neilvandyke** [gyscos/Cursive#235]
# Problem description

In an `EditView` that has scrolled right to accommodate input longer than the width of the view, pressing left-arrow in the first column of the view, in an attempt to scroll left, results in an immediate panic, due to an error such as:

```
thread 'main' panicked at
  'begin <= end (7 <= 6) when slicing `abcdefghijklmnopqrstuvwxyz`',
  libcore/str/mod.rs:2225:5
```

I can reproduce this panic consistently on my system, by: in an rxvt-unicode, using Cursive's `edit.rs` example, enter a 26-character string (the alphabet), then press the left-arrow key repeatedly until panic.

It will exit and restore the shell's terminal screen as soon as left-arrow is pressed when the cursor is in column 0 of the view, requiring a scroll.  Perhaps a fencepost error.  Exit status of the process is 101.

Attached is file [stderr.txt](https://github.com/gyscos/Cursive/files/1969813/stderr.txt) (renamed from `stderr`) from the following commands, which includes a Rust backtrace.

```
$ stty -a | head -1
speed 38400 baud; rows 24; columns 80; line = 0;

$ git log -1 | head -1
commit 3731b7375d3929eace090326c6c85e0d3a438fb4

$ RUST_BACKTRACE=1 cargo run --example edit 2> stderr

$ echo $?
101
```

# Environment

* Operating system: Debian 9.4 stable
* Backend used: ncurses 6.0+20161126-1+deb9u2
* Current locale: en_US.UTF-8
* Cursive version: git commit 3731b7375d3929eace090326c6c85e0d3a438fb4

(Thank you; Cursive is a nice library.)

**Comment by gyscos** [comment]
Hi, and thank you very much for the report!

Indeed, that was a pretty bad bug, you'd wonder why I didn't catch this earlier...

It should be fixed in d4ced015f62ed3a3bce502cdffa06f486d57b5e1.

Thanks again!

**Comment by gyscos** [comment]
Fix was released as part of 0.9.0.
